### PR TITLE
Completing user related queries and mutations

### PR DESCRIPTION
**Pull request by cl3t0**
### 💡 Summary

This PR aims to adjust the User GraphQL type and make necessary changes in the database.

### 📝 Details

- [X] Add `name` and `payday` columns to the user table;
- [X] Create an integration table that stores all user's integrations;
- [X] Add related fields in the GraphQL interface;
- [X] Change the JWT Token content to something more useful;
- [X] Verify if the token is too old or not.
- [X] Add migration to send login code;

### 🔎 References

base branch: develop

### CODE REVIEW

File: src/models/user_integration.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+impl UserIntegration {
+    fn to_service(&self) -> user_service::UserIntegration {
+        user_service::UserIntegration {
+            id: self.id.clone(),
+            related_user: self.related_user.clone(),
+            name: self.name.clone(),
+            time: self.time,
+        }
+    }
+}
```
Comment (davisuga) [inline src/models/user_integration.rs]: Don't you think that the service should be aware of the model but not vice versa? You can implement this in the service layer.

File: src/models/user_integration.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+pub type Result<T> = std::result::Result<T, IntegrationModelError>;
+
+pub fn create_integration(
+    conn: &database::DbPool,
+    t: user_service::NewUserIntegration,
+) -> Result<user_service::UserIntegration> {
+    diesel::insert_into(user_integration_schema::table)
+        .values(&t.to_model())
+        .get_result::<UserIntegration>(&mut conn.get()?)
+        .map(|t| t.to_service())
```
Comment (davisuga) [inline src/models/user_integration.rs]: You don't need to create this kind of conversion in the model layer, you just need to query the database and return the results. Don't you think that this kind of conversion may include some business logic?

File: src/services/user.rs [hunk 3]
```diff
@@ -96,10 +176,41 @@ pub fn validate_and_generate_token(
...
 }
+
+pub fn create_integration(
+    conn: &database::DbPool,
+    token: &str,
+    jwt_secret: &str,
+    name: String,
+    time: NaiveDateTime,
+) -> Result<UserIntegration> {
+    let id = jwt::verify_token(Utc::now().naive_utc(), token, jwt_secret)?;
```
Comment (davisuga) [inline src/services/user.rs:138]: Can't you verify the token on the "upper" layer, as middleware? This is kinda repetitive here.

**Review by davisuga — CHANGES_REQUESTED** [review]
Amazing job as always. Just a few points related to repetitive verification code and dependency direction. Note that I'm not commenting on the same thing twice when something repeats here.

File: src/models/user_integration.rs (hunk 1 above)
Comment (cl3t0) [reply src/models/user_integration.rs]: Fixed.

File: src/models/user_integration.rs (hunk 2 above)
Comment (cl3t0) [reply src/models/user_integration.rs]: Fixed.

File: src/services/user.rs (hunk 3 above)
Comment (cl3t0) [reply src/services/user.rs:138]: Fixed.

**Review by davisuga — APPROVED** [review]

**Pull request merged**